File: lessons/compiled_lang/cpp/src/f41.loop.cpp
// testbox: title="for (;;) with continue"
#include <iostream>
#include <string>

// true if s is empty or contains only whitespace
bool is_blank(const std::string& s) {
    return s.find_first_not_of(" \t\r\n") == std::string::npos;
}

int main() {
    for (;;) {
        std::cout << "Enter your name (quit to exit): ";
        std::string answer;
        std::getline(std::cin, answer);

        if (is_blank(answer))
            continue;

        if (answer == "quit")
            break;

        std::cout << "Hello " << answer << "!" << std::endl;
    }

    return 0;
}

File: lessons/compiled_lang/java/src/f41.loop.java
// testbox: title="for (;;) with continue"
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class F41Loop {
    public static void main(String[] args) throws IOException {
        BufferedReader stdin = new BufferedReader(new InputStreamReader(System.in));
        for (;;) {
            System.out.print("Enter your name (quit to exit): ");
            String answer = stdin.readLine();

            if (answer.isBlank())
                continue;

            if (answer.equals("quit"))
                break;

            System.out.println("Hello " + answer + "!");
        }
    }
}
